reactor-core/src/main/java/reactor/core/scheduler/ExecutorServiceWorker.java

```diff
@@ -31,9 +32,12 @@
 
 	final Composite tasks;
 
-	ExecutorServiceWorker(ScheduledExecutorService exec) {
+	final WeakReference<Scheduler> scheduler;
```

Comment:
not sure about that @smaldini, for it to be useful I had to link back to the `Scheduler`. wdyt?

Reply:
actually, this can probably also benefit of `Schedulers.scanExecutor` (which I added later)

Reply:
I think we don't need to back reference Scheduler necessarily. Maybe just use ScheduledExecutorService ?